harness/determined/common/experimental/resource_pool.py

```diff
@@ -0,0 +1,113 @@
+from typing import List, Optional
+
+from determined.common import api
+from determined.common.api import bindings
+from determined.common.experimental import workspace
+
+
+class ResourcePool:
+    """A class representing a resource pool object.
+
+    Attributes:
+        name: (str) The name of the resource pool.
+    """
+
+    def __init__(
+        self,
+        session: api.Session,
+        name: Optional[str] = None,
+    ):
+        """Create a resource pool object.
+
+        Arguments:
+            session: The session to use for API calls.
+            workspace_name: (Optional) Name of the resource pool.
+        """
+        self.name = name
+        self._session = session
+
+    @classmethod
+    def _from_bindings(
+        cls, resource_pool_bindings: bindings.v1ResourcePool, session: api.Session
+    ) -> "ResourcePool":
+        resource_pool = cls(session, name=resource_pool_bindings.name)
+        return resource_pool
+
+    def bind(
```

Comment:
The paths within the CLI have recently undergone an evolution, and I think it could make sense to mirror them here with `add_binding`, `remove_binding`, and `replace_binding`. But I also like this interface, too.

Just wanted to bring in the most up-to-date information possible so you could make whatever decision you most prefer. I think both sets of names are great.